### ISSUE TITLE
Err on a linux with `ipv6.disable=1`: `Protocol not available (os error 92)`

### ISSUE THREAD

**Issue by honwen** [EAimTY/tuic#164]
整个linux环境是 ipv4 only的

https://github.com/EAimTY/tuic/blob/f6602beaf849e57c984e1e1c7c08110fdd328025/tuic-server/src/server.rs#L101

不是很明白这里逻辑，这里是`dual_stack=false` 等价于 ipv6_only ?
注释掉这部分能正常启动server

server-config:
```json
{
  "server": "0.0.0.0:443",
  "users": {
    "00000000-0000-0000-0000-000000000000": "PASSWORD_0",
    "00000000-0000-0000-0000-000000000001": "PASSWORD_1"
  },
  "certificate": "example/example.dev.csr",
  "private_key": "example/example.dev.key",
  "congestion_control": "cubic",
  "alpn": ["h3", "spdy/3.1"],
  "udp_relay_ipv6": false,
  "dual_stack": false
}

```

**Comment by honwen** [comment]
https://github.com/EAimTY/tuic/blob/f6602beaf849e57c984e1e1c7c08110fdd328025/tuic-client/src/socks5.rs#L46

https://github.com/EAimTY/tuic/blob/f6602beaf849e57c984e1e1c7c08110fdd328025/tuic-client/src/socks5.rs#L122

这几部分代码都注释后, CS间能正常运作

```json
{
  "relay": {
    "server": "example.dev:443",
    "uuid": "00000000-0000-0000-0000-000000000000",
    "password": "PASSWORD_0",
    "ip": "127.0.0.1",
    "udp_relay_mode": "native",
    "congestion_control": "cubic",
    "alpn": ["h3", "spdy/3.1"]
  },
  "local": {
    "server": "0.0.0.0:1080",
    "dual_stack": false
  }
}
```

**Comment by EAimTY** [comment]
这个错误应该是由在 IPv4 socket 上设定 dual-stack socket 导致的，f0e931d7bd8eb38da428188d43594c91b2ba234b 与 f3cb84e5c47e70edeb32a3e6688e8f81ac539671 增加了更多相关的错误提示

**Comment by honwen** [comment]
tested ok. tks 4 Ur patience

**Comment by EAimTY** [comment]
> According to https://github.com/rust-lang/socket2/pull/260 , a socket should be set to only_v6 before binding it to an address. This caused the problem still occured in 1.0.0-pre-alpha1

now this should be fixed in 5dd3c920e49122e6a62368b0d9b919839550baae
